poetry/puzzle/provider.py

```diff
@@ -182,34 +181,34 @@ def search_for_vcs(self, dependency):  # type: (VCSDependency) -> List[Package]
 
     @classmethod
     def get_package_from_vcs(
-        cls, vcs, url, reference=None, name=None
-    ):  # type: (str, str, Optional[str], Optional[str]) -> Package
+        cls, vcs, url, reference=None, name=None, subdirectory=None
+    ):  # type: (str, str, Optional[str], Optional[str], Optional[str]) -> Package
         if vcs != "git":
             raise ValueError("Unsupported VCS dependency {}".format(vcs))
 
-        tmp_dir = Path(
-            mkdtemp(prefix="pypoetry-git-{}".format(url.split("/")[-1].rstrip(".git")))
-        )
+        tmp_dir = Path(DownloadTmpDir.mkcache(url, prefix="pypoetry-git"))
```

Comment:
This this is not longer required, since this has changed:

https://github.com/python-poetry/poetry/blob/063d19aa7d5e9b8601a2658237dfc70995410c7e/poetry/puzzle/provider.py#L190-L192

Reply:
Hm, don't know if you link the correct lines?

The reason for the way I create the temp folder for cloning the repository is, that at the moment poetry clone the repository twice during install. The first time to resolve the package version and second time when installing. For the case, let's say I like to add 3 packages from subdirectory of the same repo, poetry would clone this repo 6 times. With the newly added class `DownloadTmpDir` it will clone the repository only once during runtime and reuse it for every package that is clone from the same repository.

Reply:
Yeah, you can ignore the snippet. I understand the intention here now.

Do you know if there was a reason why we need a temporary clone here at all? Can we for example, make this not runtime specific? Clone a repo to a deterministic source directory (maybe a name containing `hash(source_url)`) and switch revisions instead of a new clone? This should avoid cloing across runtimes - might be advantageous in non subdirectory use case as well. If so, I would say it might even be a separate PR independant from the subdirectory change as it has broader impact.